torch/export/serialization.py

```diff
@@ -0,0 +1,311 @@
+            for user_node in fx_node.users:
+                assert user_node.target is operator.getitem, "Consumer of multiple return values should be getitem"
+                output_fake_tensors[user_node.name] = user_node.meta.get("val", None)
+        else:
+            output_fake_tensors[fx_node.name] = fx_node.meta.get("val", None)
+
+
+        if isinstance(fx_node.target, torch._ops.OpOverload):
+            target_name = str(fx_node.target)
+        else:
+            target_name = str(fx_node.target)
+
+        node = ex.Node(
+            op = fx_node.op,
+            target = target_name,
+            args = export_node_args(fx_node.args),
+            kwargs = export_node_kwargs(fx_node.kwargs),
+            outputs = [
+                ex.TensorArgument(
+                    name = name,
+                ) for name in output_fake_tensors.keys()
+            ],
+            # TODO: create a new ivalue here, meta might have faketensor info
+            metadata = export_node_meta(self.current_node.meta),
+        )
+        self.ex_graph.nodes.append(node)
+
+        for name, fake_tensor in output_fake_tensors.items():
+            ivalue = ex.IValue(
+                name = name,
+                meta = export_tensor_meta(fake_tensor) if fake_tensor is not None else None,
+            )
+            self.ex_graph.ivalues.append(ivalue)
+
+        return result
+
+
+    def get_attr(self, target, args, kwargs):
+        result = super().get_attr(target, args, kwargs)
+        # TODO: assert target is in ex_gm.parameters or buffers
+        return result
+
+
+    def call_module(self, target, args, kwargs):
+        raise AssertionError()
+
+    def call_method(self, target, args, kwargs):
+        raise AssertionError()
+
+    def output(self, target, args, kwargs):
+        result = super().output(target, args, kwargs)
+
+        assert len(args) > 0, "output node should have at least one arg"
+        assert isinstance(result, (tuple, list)), type(result)
+
+        fx_node = self.current_node
+        assert len(fx_node.args) == 1, "fx_node's args should have one arg"
+        node_args = fx_node.args[0]
+
+        node = ex.Node(
+            op = fx_node.op,
+            target = fx_node.target,
+            args = [
+                ex.TensorArgument(
+                    name = str(arg),
+                ) for arg in node_args
+            ],
+            kwargs = [],
+            outputs = [
+                ex.TensorArgument(
+                    name = str(arg),
+                ) for arg in node_args
+            ],
+            metadata = export_node_meta(self.current_node.meta),
+        )
+
+        self.ex_graph.output = node
+
+        return result
+
+
+device = "cuda"
+batch_size = 2
+model = resnet18().cuda().eval()
+x = torch.rand(batch_size, 3, 224, 224, device=device, dtype=torch.float)
+gm, guard = torch._dynamo.export(model, x, aten_graph=True)
+
+exporter = ExportInterpreter(gm)
+exporter.run(x)
```

Comment:
I would be willing to accept this rn, but in the long run we should definitely not use the Interpreter pattern for serialization; in particular, it doesn't make sense to require that we provide inputs again to serialize the program.

Reply:
For dynamo.export, we need a concrete data anyway. We can reuse the data provided to dynamo.export. 

I don't have strong opinions on whether it should follow the Interpreter pattern, but it's the easiest for prototype :)